### ISSUE TITLE
Roadmap / release planning

### ISSUE THREAD

**Issue by gnutix** [Imagick/imagick#297]
Hello. I'm currently redeveloping a huge application on PHP 7.3 and am facing issues because of old Imagick binaries (no webp support for example). I've tried to compile IM 7 + php-imagick but failed.

So I was wondering what was the planning for the release of "the next php-imagick version" (that will finally be shipped with IM7 / PHP7) ? Weeks, months, years ?

Thanks for the insight.

**Comment by Danack** [comment]
A 'good news, bad news' situation.

Good news - you don't need to wait for the next release to solve that problem, you can fix it yourself.

Bad news - you'll have to fix it yourself, and so you'll need to get ImageMagick compiling.

The short version is that Imagick doesn't have anything to do with what image formats are available. Also, ImageMagick only picks up which image formats can be used when it is recompiled. Which


I should probably write up the instructions on how to do this properly, but you can look at this script to do it, and related docker image: https://github.com/Imagick/imagick/blob/master/docker/installImageMagick.sh


You could also speak to whoever looks after the ImageMagick package for your platform. tbh, adding webp support by default is probably a good idea.

**Comment by Danack** [comment]
Oh wait....to clarify one thing:

> that will finally be shipped with IM7 / PHP7

The current releases support the latest versions of those...so I don't know precisely what you're asking there.

**Comment by gnutix** [comment]
Hey @Danack, thanks for answering back. Well, I do have `cwebp` and `gif2webp` installed on my machine.

Our application has currently three "passes" to process images : 1) create thumbnails using `convert`. 2) create `.webp` files using `cwebp`/`gif2webp`. 3) optimize the images with various tools.

I'd like to be able to reduce this to two steps only, has `convert` should be able to create `webp` images (by "delegating", AFAIK). 

What is strange, is that I had these two binaries when I tried to compile `ImageMagick 7` and re-compile `php-imagick 3.4.4`. And yet, when trying a simple `convert image.jpg image.webp`, it didn't work. As I had just worked for many hours in a go, I dropped it and created this issue for followthrough.

Maybe I was missing some compilation flags ? I remember calling `./configure` without arguments, and looking at your Docker scripts there are flags like `--with-jpeg=yes` and `--with-png=yes`.

Is there one for `webp` support ? Will it delegate only to `cwebp` or also `gif2webp` ?

**Comment by Danack** [comment]
> Is there one for webp support ?

Yes. I can't remember what it is though. If you run the ImageMagick configure it lists all of the configure options (and what it's using as it's defaults) as part of the output.

> Will it delegate only to cwebp or also gif2webp ?

I have zero idea. If you can't see what it's going to do from the output of the configure command or possibly from the config.log file, you'd need to ask the ImageMagick team.

**Comment by Danack** [comment]
oh actually, the dependency ImageMagick is more likely to be going to look for is one of:

* libwebp-dev
* libwebp6

pretty sure it would be looking for that, rather than a delegate tool.
